### PR TITLE
Do not try to integrate uncreated test native targets

### PR DESCRIPTION
**Pull request by dnkoutso**
As of 1.5.2 (and will be in 1.6.0) vendored pods (`should_build?` returns `false`) will not integrate their test specs. Some bigger changes are required to accomplish this.

This fixes an edge case in which a pod target must be integrated because it has script phases but it should not be built. The code assumes that a test target has been created for the test specs.

This fixes this edge case.

Found as part of https://github.com/CocoaPods/CocoaPods/issues/7394

### CODE REVIEW

**Comment by dnkoutso** [comment]
We should get a more comprehensive fix for which test specs are installed and integrated even for pods that have no sources but for 1.7.0 or 1.6.1.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Do not try to integrate uncreated test native targets  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/xcode/pods_project_generator/target_installation_result.rb [hunk 1]
```diff
@@ -73,9 +73,10 @@ def native_target_for_spec(spec)
           #         an array of all the test specs associated with this native target.
           #
           def test_specs_by_native_target
-            target.test_specs.group_by do |test_spec|
+            test_specs_by_native_target = target.test_specs.group_by do |test_spec|
               test_native_target_from_spec(test_spec)
             end
+            test_specs_by_native_target.delete_if { |k, _| k.nil? }
```
Comment (segiddins) [inline lib/cocoapods/installer/xcode/pods_project_generator/target_installation_result.rb]: this should have a test and also am error message that integrating test specs into pods that are not being built is not yet supported

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/target_installation_result.rb]: tests added
